okhttp/src/main/kotlin/okhttp3/internal/cache/CacheInterceptor.kt

```diff
@@ -104,7 +115,9 @@ class CacheInterceptor(internal val cache: Cache?) : Interceptor {
         // Content-Encoding header (as performed by initContentStream()).
         cache!!.trackConditionalCacheHit()
         cache.update(cacheResponse, response)
-        return response
+        return response.also {
+          listener.cacheHit(call, it)
```

Comment:
should we track conditionalCacheHits separately?

Reply:
There’s decisions about whether we got a response from the network or not, got a response from the cache or not, and which one won

Reply:
Yep - this accounting is different than the Cache counters.  I wanted to keep this model simple, but you can always look at the network and cache responses, also prior response to understand more.

But critically, without these new events, EventListeners aren't guaranteed to see a Response to work off.  So this seems pretty useful.